testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -354,7 +354,7 @@ impl EventObserver {
     fn make_new_block_processed_payload(
         &self,
         filtered_events: Vec<(usize, &(bool, Txid, &StacksTransactionEvent))>,
-        block: &StacksBlock,
+        block: StacksBlockEventData,
```

Comment:
Why not pass `block` as a reference? It doesn't seem like this will avoid copying data, because the `block` fields are all getting string serialized anyways.

Reply:
By passing by value, we ensure that the caller can't accidentally send the same events twice.

Reply:
But the caller is calling `block.clone()`, and it definitely _is_ invoking this method multiple times with the same events.

Reply:
I can change to a reference